packages/gatsby/cache-dir/navigation.js

```diff
@@ -82,6 +82,28 @@ const navigate = (to, options = {}) => {
   }, 1000)
 
   loader.loadPage(pathname).then(pageResources => {
+    // If the loaded page has a different compilation hash to the
+    // window, then a rebuild has occurred on the server. Reload.
+    if (process.env.NODE_ENV === `production` && pageResources) {
+      if (
+        pageResources.page.webpackCompilationHash !==
+        window.___webpackCompilationHash
+      ) {
+        // Purge plugin-offline cache
+        if (
```

Comment:
seems like I didn't save my comment. This feels a bit hacky IMO but good enough to go forward with this one.

IMO it feels like we need to create a gatsby-browser api for this so plugins can handle this on their behalf. Now it looks like that core & offline plugin are intertwined. Maybe just put up a TODO comment?

Reply:
I don't like intertwining Gatsby and gatsby-plugin-offline. But adding a new API with only one known use case is _also_ problematic. Worse IMO because APIs are forever while hacks can be removed pretty quickly. This seems pretty harmless all in all and if there's ever another use case that comes up for telling people when the webpack compilation hash changes, we could add an API and move this code to gatsby-plugin-offline.